crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -297,15 +301,11 @@ pub fn export_schema() -> String {
 }
 
 async fn graphql_handler(
-    State(metrics): State<Metrics>,
     ConnectInfo(addr): ConnectInfo<SocketAddr>,
     schema: axum::Extension<SuiGraphQLSchema>,
     headers: HeaderMap,
     req: GraphQLRequest,
-) -> GraphQLResponse {
-    metrics.request_metrics.inflight_requests.inc();
-    metrics.inc_num_queries();
-    let instant = Instant::now();
+) -> (axum::http::Extensions, GraphQLResponse) {
```

Comment:
Is this a known pattern? I am trying to understand how this works...

Reply:
Yeah, look at the Axum docs for response types. The return type just needs to impl IntoResponse. There's a bit of type magic happening